go/arrow/cdata/cdata_exports.go

```diff
@@ -385,7 +388,13 @@ func exportArray(arr arrow.Array, out *CArrowArray, outSchema *CArrowSchema) {
 		for i := range bufs {
 			buf := bufs[i]
 			if buf == nil || buf.Len() == 0 {
-				buffers[i] = nil
+				if i > 0 {
```

Comment:
This implies that buffer 0 is the null bitmap, which might not be true for all types (such as unions).

By the way, the hardcoded check for unions above could instead use of more generic API (such as `HasValidityBitmap` in Arrow C++).

Reply:
@lidavidm There's a `HasValidityBitmap` in `arrow/internal/utils.go` you can import `github.com/apache/arrow/go/v12/arrow/internal` and use `internal.DefaultHasValidityBitmap`

Reply:
Updated.